crates/rune/src/index.rs

```diff
@@ -424,6 +436,12 @@ impl Index<ast::Expr> for Indexer<'_> {
     fn index(&mut self, expr: &ast::Expr) -> CompileResult<()> {
         let span = expr.span();
         log::trace!("Expr => {:?}", self.source.source(span));
+        if expr.has_attributes() {
```

Comment:
This is a pretty fine solution for now, but will be pretty short lived once we start gradually introducing attribute semantics to various types 🙃. Maybe rename it `has_unsupported_attributes` or some such and we can gradually remove variants from it once they gain support in the compiler.